openml/datasets/functions.py

```diff
@@ -897,6 +901,46 @@ def edit_dataset(
     return int(data_id)
 
 
+def fork_dataset(data_id: int) -> int:
+    """
+     Creates a new dataset version, with possibly a different owner.
+     The authenticated user becomes the new owner.
+     This function does not result in any change in the dataset ARFF file.
```

Comment:
This might be a bit low level for function documentation?
Perhaps instead something like
  "The forked dataset can have distinct dataset meta-data, but the actual data itself is shared with the original version."?